### ISSUE TITLE
Log processing rules randomly failing to be applied with autodiscovery

### ISSUE THREAD

**Issue by lluzza** [DataDog/datadog-agent#6201]
**Output of the info page (if this is a bug)**
```
$ kubectl exec -n monitoring -ti datadog-agent-worker-hzrnw -c datadog-agent s6-svstat /var/run/s6/services/agent/
up (pid 380) 64226 seconds

```

**Describe what happened:**
Trying to apply a log processing rule using autodiscovery ("ad_identifiers") and configmap in our k8s cluster works randomly depending on what the agent do first: start tailing the log file or loading the config.

If the agent loads the log config rule before start tailing the log file then the rule is applied and everything works fine but if it's all the way around the log rules are not applied:

```

agent status:
 janus-config
------------
Type: docker
Status: Pending


/var/log/datadog/agent.log:
2020-07-27 21:06:58 UTC | CORE | INFO | (pkg/logs/scheduler/scheduler.go:77 in Schedule) | Received a new service: docker://a972c27fa02b6dfa5f8cb5e16644863960b42fb7afa7ee530f3c1331ca4e4b75
.
.
.
2020-07-27 21:06:59 UTC | CORE | INFO | (pkg/logs/scheduler/scheduler.go:58 in Schedule) | Received a new logs config: janus-config

log file in /var/log/containers: janus-qa-honeycomb-java-chart-6668595885-5w6jb_honeycomb-qa_application-a972c27fa02b6dfa5f8cb5e16644863960b42fb7afa7ee530f3c1331ca4e4b75.log
```

Here you can see that the agent started tailing the log file which the log processing rule configuration should be applied to before the `janus-config` was loaded so the rules are not applied.

Restarting the agent (killing the pod) sometimes work but it seems to be a completely random process, the agent starts tailing log files and loading configurations without any order so it's just a matter of luck to get the log rules applied.

**Describe what you expected:**

The autodiscovery log rules should be loaded by the agent before start tailing log files. Here's an example when the log configuration is loaded before the log file so the log rules were applied successfully:

```
 janus-config
------------
Type: docker
Status: OK
Inputs: cb378c9c24f3d5a321a382c6bd77bcdeda09dea9030210b5a2b43f364139d0f6

/var/log/datadog/agent.log:
2020-07-27 21:15:38 UTC | CORE | INFO | (pkg/logs/scheduler/scheduler.go:58 in Schedule) | Received a new logs config: janus-config
2020-07-27 21:15:38 UTC | CORE | INFO | (pkg/logs/scheduler/scheduler.go:77 in Schedule) | Received a new service: docker://cb378c9c24f3d5a321a382c6bd77bcdeda09dea9030210b5a2b43f364139d0f6

log file in /var/log/containers: janus-dev-honeycomb-java-chart-6c99d8fbc5-v92b2_honeycomb-dev_application-cb378c9c24f3d5a321a382c6bd77bcdeda09dea9030210b5a2b43f364139d0f6.log
```

**Steps to reproduce the issue:**
Configure log rules using a configmap in k8s and autodiscovery with "ad_identifiers"

**Additional environment details (Operating System, Cloud provider, etc):**

configmap used by agent 

```
  janus-config: |-
    ad_identifiers:
      - janus
    logs:
      - type: docker
        source: janus
        service: janus
        log_processing_rules:
          type: multi_line
          name: log_start_with_time_janus
          pattern: ".*(0[0-9]|1[0-9]|2[0-3]):[0-5][0-9]:[0-5][0-9]"
```
Datadog agent: 7.21.1 - Cloud provider: AWS 

Now we are using cluster-agent but the issue was also happening when using only node-based agents.

[agent_status.txt](https://github.com/DataDog/datadog-agent/files/5059857/agent_status.txt)

**Comment by JM-Lazaro** [comment]
Hey, thanks for reporting this! We have created an internal ticket for further investigating this and to develop a fix. 
For the meantime, you can use apply the logs config on your target pods using annotations as a workaround. Please check out our [documentation](https://docs.datadoghq.com/agent/kubernetes/log/?tab=kubernetes#configuration) for details.

**Comment by lluzza** [comment]
> Hey, thanks for reporting this! We have created an internal ticket for further investigating this and to develop a fix.
> For the meantime, you can use apply the logs config on your target pods using annotations as a workaround. Please check out our [documentation](https://docs.datadoghq.com/agent/kubernetes/log/?tab=kubernetes#configuration) for details.

Thanks JM! We already tried annotations to apply log rules (mainly multiline) and they worked fine but we are looking for a solution to this issue that does not imply any change to our pod deployments. We have a rather complex process to deploy to prod so we would like to keep monitoring configurations as isolated from applications as possible. 

Regards,
Luis.

**Comment by togikirankumar** [comment]
Hey @JM-Lazaro is this issue fixed now. I am also facing the same issue.

**Comment by JM-Lazaro** [comment]
@togikirankumar Hey, yes, it was fixed in Agent 7.23 through this PR: https://github.com/DataDog/datadog-agent/pull/6316

Could you please check if you are using Agent 7.23 or higher? If not, please try upgrading your agent. 
Should the issue persist, or if you were using a higher version of the agent already, we'll likely need to gather more information and the best way to share those would be through a support ticket. You can create one by sending an email to support@datadoghq.com.

Let me know if you have any questions.

**Comment by JM-Lazaro** [comment]
As per my last comment, the fix is included in Datadog Agent 7.23.0 or higher.
